homeassistant/components/ipp/__init__.py

```diff
@@ -41,29 +41,33 @@
 
 async def async_setup(hass: HomeAssistant, config: Dict) -> bool:
     """Set up the IPP component."""
-    hass.data.setdefault(DOMAIN, {})
+    if DOMAIN not in hass.data:
```

Comment:
This line is not needed, as `setdefault` doesn't harm existing values (it will apply the default if the key isn't in the dictionary already).

Reply:
sorry, wasn't aware of that